### PR TITLE
refactor(node): reimplement http client

### PR DESCRIPTION
**Pull request by crowlKats**
This commit reimplements most of "node:http" client APIs using "ext/fetch".

There is some duplicated code and two removed Node compat tests that will
be fixed in follow up PRs.

### CODE REVIEW

File: ext/node/ops/http.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+use reqwest::header::HeaderName;
+use reqwest::header::HeaderValue;
+use reqwest::header::CONTENT_LENGTH;
+use reqwest::Body;
+use reqwest::Method;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+#[op]
+pub fn op_node_http_request(
```
Comment (ry) [inline ext/node/ops/http.rs:33]: Maybe add a comment to duplicate with op_fetch from https://github.com/denoland/deno/blob/0ccfccdcd271cb75f7453c70c6c3f4cadd7f3858/ext/fetch/lib.rs#L215

Comment (lucacasonato) [inline ext/node/ops/http.rs:33]: Why is this duplicate op needed?

Comment (crowlKats) [reply ext/node/ops/http.rs:25]: the original fetch op filters out some set header values. This op could most likely be deduplicated

Comment (bartlomieju) [reply ext/node/ops/http.rs:27]: We'll deduplicate in a follow up PR

Comment (bartlomieju) [reply ext/node/ops/http.rs:27]: We'll deduplicate in a follow up PR

**Review by bartlomieju — APPROVED** [review]
LGTM, we will iterate on this module a lot more this week, but this is already a huge improvement.

Next step: make sure that `ext/node` only uses HTTP 1.1.

**Pull request merged**